flask_api/app.py

```diff
@@ -29,14 +36,54 @@ def parse_user_id(req):
     print(req.headers)
     return req.headers['Authorization'].split()[1]
 
+def resize_n_upload(original_file, New_file, date_time, s3_key_name):
+    s3_client = boto3.client('s3')
+    s3_bucket_name = "image-bucket-paul"
+
+    im = Image.open(original_file)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *im*. The resource is allocated by call *Image.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *im* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).